sql/core/src/test/scala/org/apache/spark/sql/StringLiteralCoalescingSuite.scala

```diff
@@ -214,13 +214,13 @@ class StringLiteralCoalescingSuite extends QueryTest with SharedSparkSession {
   test("R-string coalescing with regular strings") {
     val testCases = Seq(
       // R-string followed by regular string
-      ("""SELECT R'\n' ' tab'""", """\n tab"""),
+      ("""SELECT R'\n' ' tab'""", raw"\n tab"),
       // Regular string followed by R-string
-      ("""SELECT 'newline ' R'\n'""", """newline \n"""),
+      ("""SELECT 'newline ' R'\n'""", raw"newline \n"),
```

Comment:
what's the coalesced token for `SELECT 'newline ' R'\n'`? it becomes `R'newline \n'`? then what if the normal string literal has escaping inside it?